### PR TITLE
pad/pdn: improve bump handling

### PR DESCRIPTION
**Pull request by gadfort**
Adds:
- dont_route to `assign_io_bump` command to allow for assignment of bumps without attempting to route the bump
- `-bump` flag to pdn macro grid definition to skip adding grids to bumps that have already been routed to.

### CODE REVIEW

**Comment by maliberty** [comment]
What is the use case? Pre routing?

File: src/pad/src/RDLRouter.cpp [hunk 1]
```diff
@@ -1152,6 +1152,8 @@ std::vector<RDLRouter::TargetPair> RDLRouter::generateRoutingPairs(
   odb::dbTechLayer* bump_pin_layer = get_other_layer(bump_accessvia_);
   odb::dbTechLayer* pad_pin_layer = get_other_layer(pad_accessvia_);
 
+  const double dbus = block_->getDbUnitsPerMicron();
```
Comment (maliberty) [inline src/pad/src/RDLRouter.cpp:1155]: This looks to be defined far from it's use

**Comment by gadfort** [comment]
> What is the use case? Pre routing?

The bumps in the center of chip cannot be routed with the RDL router. This still allows them to be correctly assigned (ie. a bpin is created and the bump pins are connected to the correct net), but the RDL router will not attempt to route them. PDNgen can then add a grid around them to connect to the bump properly.

File: src/pad/src/RDLRouter.cpp (hunk 1 above)
Comment (gadfort) [reply src/pad/src/RDLRouter.cpp:1155]: I can move it down closer.

**Review by maliberty — APPROVED** [review]

**Pull request merged**